File: WsServer/WsServer.Shared/Rooms/CommunicationMode.cs
namespace WsServer.Rooms;

/// <summary>
/// Defines supported communication modes for rooms
/// </summary>
public enum CommunicationMode
{
    /// <summary>
    /// 2D spatial mode with position updates
    /// </summary>
    Spatial2D = 1,
    
    /// <summary>
    /// 3D spatial mode with position updates
    /// </summary>
    Spatial3D = 2,
    
    /// <summary>
    /// Text chat only mode
    /// </summary>
    TextChat = 3,
    
    /// <summary>
    /// Voice chat only mode
    /// </summary>
    VoiceChat = 4,
    
    /// <summary>
    /// Video chat only mode
    /// </summary>
    VideoChat = 5
}

File: WsCore.Client/src/network/rooms/CommunicationMode.ts
/**
 * Defines supported communication modes for rooms
 */
export enum CommunicationMode {
  /**
   * Spatial mode with real-time position updates (the 3D game room).
   * Maps to the server's single RoomCompatibility.Spatial.
   */
  Spatial = 'spatial',

  /**
   * Text chat only mode
   */
  TextChat = 'text_chat',

  /**
   * Voice chat only mode
   */
  VoiceChat = 'voice_chat',

  /**
   * Video chat only mode
   */
  VideoChat = 'video_chat',
}
